examples/single_column_carbon_alkalinity_nutrients.jl

```diff
@@ -20,118 +22,398 @@ grid = RectilinearGrid(size = 64,
+
+# For air-sea CO₂ fluxes, we use a FluxBoundaryCondition for the "top" of the DIC tracer.
+# We'll write a callback to calculate the flux every time step.
+co₂_flux = Field{Center, Center, Nothing}(grid)
 
-N₀ = 1e-1 # Surface nutrient concentration
-D₀ = 1e-1 # Surface detritus concentration
-dᴺ = 50.0 # Nutrient mixed layer depth
-N² = 1e-5 # Buoyancy gradient, s⁻²
+dic_bcs  = FieldBoundaryConditions(top = FluxBoundaryCondition(co₂_flux))
 
-bᵢ(z) = N² * z
-Nᵢ(z) = N₀ * max(1, exp(-(z + dᴺ) / 100))
-Dᵢ(z) = D₀ * exp(z / 10)
+### We put the pieces together ###
+# The important line here is `biogeochemistry = CarbonAlkalinityNutrients()`.
+# We use all default parameters.
+model = HydrostaticFreeSurfaceModel(; 
+    grid,
+    biogeochemistry     = CarbonAlkalinityNutrients(; 
+        grid,
+        ),
+    closure             = ScalarDiffusivity(; κ=1e-5),    
+    tracers             = (:T, :S, :DIC, :ALK, :PO₄, :NO₃, :DOP, :POP, :Fe),
+    tracer_advection    = WENO(),
+    buoyancy            = SeawaterBuoyancy(
+        equation_of_state  = LinearEquationOfState(
+            thermal_expansion  = αᵀ,
+            haline_contraction = βˢ
+            ),
+        ),
+    boundary_conditions = (; T=T_bcs, DIC=dic_bcs),
+    )
 
-# set!(model, b=bᵢ, P=1e-1, B=1e-1, D=Dᵢ, N=Nᵢ, e=1e-6)
-set!(model, b=bᵢ, e=1e-6)
+# ## Initial conditions
+#
+## Temperature initial condition: a stable density gradient with random noise superposed.
+## Random noise damped at top and bottom
+Ξ(z) = randn() * z / model.grid.Lz * (1 + z / model.grid.Lz)
+Tᵢ(z) = 20 + dTdz * z + dTdz * model.grid.Lz * 1e-6 * Ξ(z)
+Sᵢ(z) = 35
+
+# We initialize the model with reasonable nutrient concentrations
+DICᵢ(z) = 2.1   # mol/m³ 
+ALKᵢ(z) = 2.35   # mol/m³ 
+PO₄ᵢ(z) = 2.5e-3 # mol/m³ 
+NO₃ᵢ(z) = 36e-3  # mol/m³ 
+DOPᵢ(z) = 0.0    # mol/m³ 
+POPᵢ(z) = 0.0    # mol/m³ 
+Feᵢ(z)  = 1.e-6  # mol/m³
+
+set!(
+    model, 
+    T   = Tᵢ,
+    S   = Sᵢ, 
+    DIC = DICᵢ, 
+    ALK = ALKᵢ, 
+    PO₄ = PO₄ᵢ, 
+    NO₃ = NO₃ᵢ, 
+    DOP = DOPᵢ, 
+    POP = POPᵢ, 
+    Fe  = Feᵢ
+    )
 
 # ## A simulation of physical-biological interaction
 # 
 # We construct a simple simulation that emits a message every 10 iterations
 # and outputs tracer fields.
 
-simulation = Simulation(model, Δt=10minutes, stop_time=12days)
+simulation = Simulation(model, Δt=10minutes, stop_time=30days)
+
+# We add a `TimeStepWizard` callback to adapt the simulation's time-step,
+wizard = TimeStepWizard(cfl=0.2, max_change=1.1, max_Δt=60minutes)
+simulation.callbacks[:wizard] = Callback(wizard, IterationInterval(10))
+
+
+DIC₀ = Field{Center, Center, Nothing}(grid)
+ALK₀ = Field{Center, Center, Nothing}(grid)
+"""
+Function to print DIC, ALK, and CO₂ flux tendency during the simulation
+"""
+function progress(simulation) 
+    @printf("Iteration: %d, time: %s\n", 
+        iteration(simulation), 
+        prettytime(simulation),
+        )
+
+    if (iteration(simulation) == 1)
```

Comment:
```suggestion
    if iteration(simulation) == 1
```

the parentheses don't have meaning there